### ISSUE TITLE
App data clear API

### ISSUE THREAD

**Issue by d4vidi** [wix/Detox#2537]
Add support for Application clear-data, so that apps wont have to be reinstalled on Android (clearing the data is much faster).

**Comment by LeoNatan** [comment]
Isn't this supported by ADB? I thought we are already using it.

**Comment by d4vidi** [comment]
Sort of, but it's not as simple as running a single command.
In any case, we might want to consider an API that would do this seamlessly on Android and iOS (e.g. a `resetApp()` call that would clear data on android, and reinstall on iOS)... does that make sense?

**Comment by LeoNatan** [comment]
Reinstall in iOS doesn’t guarantee data clearance in some cases. I prefer not to offer API that cannot do what the users explicitly is asking for.

**Comment by d4vidi** [comment]
Yes you are right. My aim was to try to avoid spreading `if (isIos) device.launchApp({delete: true}) else device.clearData()` being implemented over and over out there.

**Comment by d4vidi** [comment]
This will be introduced indirectly via #2534

**Comment by noomorph** [comment]
The #3497 did not enter the master, so I think it is totally fine to keep this issue open for a while.
